src/tribler/gui/network/request_manager.py

```diff
@@ -0,0 +1,150 @@
+
+        # A workaround for Qt5 bug. See https://github.com/Tribler/tribler/issues/7018
+        self.setNetworkAccessible(QNetworkAccessManager.Accessible)
+
+        request.reply = self.sendCustomRequest(qt_request, request.method.encode("utf8"), buf)
+        buf.setParent(request.reply)
+
+        connect(request.reply.finished, request._on_finished)  # pylint: disable=protected-access
+
+    def remove(self, request: Request):
+        self.active_requests.pop(request, None)
+
+        if request.reply:
+            request.reply.deleteLater()
+            request.reply = None
+
+    def update(self, request: Request, status: int):
+        self.logger.debug(f'Update {request}: {status}')
+
+        self.performed_requests[request] = status
+
+    def show_error(self, request: Request, data: Dict) -> str:
+        text = self.get_message_from_error(data)
+        if self.window.core_manager.shutting_down:
+            return ''
+
+        text = f'An error occurred during the request "{request}":\n\n{text}'
+        error_dialog = ConfirmationDialog(self.window, "Request error", text, [('CLOSE', BUTTON_TYPE_NORMAL)])
+
+        def on_close(_):
+            error_dialog.close_dialog()
+
+        connect(error_dialog.button_clicked, on_close)
+        error_dialog.show()
+        return text
+
+    def _get_base_url(self) -> str:
+        return f'{self.protocol}://{self.host}:{self.port}/'
+
+    @staticmethod
+    def get_message_from_error(d: Dict) -> str:
+        error = d.get('error', {})
+        if isinstance(error, str):
+            return error
+
+        if message := error.get('message'):
+            return message
+
+        return json.dumps(d)
+
+    def clear(self, skip_shutdown_request=True):
+        for req in list(self.active_requests.values()):
+            if skip_shutdown_request and isinstance(req, ShutdownRequest):
+                continue
+            req.cancel_request()
+
+    def _drop_timed_out_requests(self):
+        for req in list(self.active_requests.values()):
+            is_time_to_cancel = time() - req.time > self.timeout_interval
+            if is_time_to_cancel:
+                req.cancel()
+
+    def _urlencode(self, data):
+        # Convert all keys and values in the data to utf-8 unicode strings
+        utf8_items = []
+        for key, value in data.items():
+            if isinstance(value, list):
+                utf8_items.extend([self._urlencode_single(key, list_item) for list_item in value if value])
+            else:
+                utf8_items.append(self._urlencode_single(key, value))
+
+        data = "&".join(utf8_items)
+        return data
+
+    @staticmethod
+    def _urlencode_single(key, value):
+        utf8_key = quote_plus(str(key).encode('utf-8'))
+        # Convert bool values to ints
+        if isinstance(value, bool):
+            value = int(value)
+        utf8_value = quote_plus(str(value).encode('utf-8'))
+        return f"{utf8_key}={utf8_value}"
```

Comment:
Instead of these methods, it is possible to use the standard `urlencode` function with the same result. While doing this, it is necessary to ensure that `True`/`False` values are encoded as `1`/`0`, but it is easy to do.